lib/Entities/SegmentEntity.php

```diff
@@ -43,6 +44,19 @@ class SegmentEntity {
    */
   private $description;
 
+  /**
+   * @ORM\OneToMany(targetEntity="MailPoet\Entities\DynamicSegmentFilterEntity", mappedBy="segment")
+   * @var DynamicSegmentFilterEntity[]|ArrayCollection
```

Comment:
Dynamic segments and segments share the same entity for now. We could use [single table inheritance](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/inheritance-mapping.html#single-table-inheritance) to create a different entities for static and dynamic segments. I think it is a bit overkill since we have only one additional property. I also noticed that in some places in the old code we kind of expect that there might be some unknown segment type and this may cause issues with @DiscriminatorMap since it must contain complete list.
I also think it is not a big deal if Segment which is not dynamic has an empty list of filters.